### ISSUE TITLE
Feature request: 'Escape' key should always close an empty popup menu

### ISSUE THREAD

**Issue by alon91** [foray1010/Popup-my-Bookmarks#160]
The Escape key appears to work differently for the following scenarios:

1. Open popup, hit Escape: the popup window closes
2. Open popup, type one or more characters, hit Escape: the search box is cleared
3. **Open popup, type one or more characters, hit Escape two times:** the search box is cleared on the first Escape, nothing happens on the second Escape

To close the popup window in the last scenario I have to either hit the extension's keyboard shortcut or mouse click outside the window.

Is it possible to close the popup window on the second Escape for the last scenario?

**Comment by foray1010** [comment]
it can be done by overriding escape keydown/keyup event, I will add it in next version when I have time (or PR is always welcomed)

example code:
```js
window.onkeydown = window.onkeyup = function(e) { if (e.keyCode == 27 /* ESC */) { e.preventDefault(); } };
```

**Comment by foray1010** [comment]
done in https://github.com/foray1010/Popup-my-Bookmarks/commit/7402071cb8e38fa8da5e676b4c2410a470e01703

will deploy until rewrite is completed

**Comment by alon91** [comment]
Amazing!! So excited to try this in the upcoming release -- thank you so much sir! :)

**Comment by alon91** [comment]
Hello! Big apologies for nag, but was curious if there was a possible update on timing for this feature request. Would love to be able to reliably escape out of my PMB window.

Thank you as always!

**Comment by alon91** [comment]
No apologies needed of course!

I would loooove a patch when you get a chance. I continue to use this extension dozens of times a day -- I've added hundreds of bookmarks because of it and the extension is something I can't work without at this point!

Let me know if I can help with testing, feedback, etc. Thank you as always!!

**Comment by foray1010** [comment]
5.3.0 is published :)

**Comment by alon91** [comment]
Not sure if this is my mistake, but right now with version 5.3 the escape key doesn't work under any scenario. It doesn't clear any entered characters or close the popup window (even when nothing was entered).

I'm on OS X, using Chrome 68.0.3440.106. Happy to troubleshoot or try to update my browser. Thank you as always!

**Comment by foray1010** [comment]
@alon91 oh I must be just reading the title! I have reverted the change in 5.3.1, let's discuss again!

1. `Open popup, hit Escape: the popup window closes`
2. `Open popup, type one or more characters, hit Escape two times: the search box is cleared on the first Escape, nothing happens on the second Escape`

A few people asked if I can disable the functionality of closing window by Escape. It is because they are not typing English but other languages that required `Escape` key to cancel their partial input (such as Chinese requires to input at least three characters to output one Chinese word), which may accidentally close the extension. I think it may be a similar story for those who want `Escape` to clear the text but not closing the extension.

tl;dr, the problem is the Escape key is used for multiple purposes and sometimes it triggered in an unwanted scenario, have you faced this issue before?

**Comment by alon91** [comment]
Thanks for the explanation, @foray1010! 

To clarify my feature request: it would be great if the `Escape` key always closed the popup window. Sometimes I search for a bookmark and realize I don't have one. When this happens I need to close the popup window by either 1) using my mouse, which I try to avoid, or 2) using the keyboard shortcut that opened the popup.

It would be great if the `Escape` key always closed the window! It is a very universal behavior for that key. I will admit that I am not aware of how other languages work, but most other extensions I've used follow this convention.

Thank you as always. PMB continues to be a life saver!

**Comment by alon91** [comment]
Hello! Quick update on this issue -- I've recently switched to using Chrome OS, which indirectly makes this problem worse. Here are details:

- I've selected `Ctrl` + `Shift` + `L` as the keyboard shortcut to activate PMB
- I've also started getting into the habit of hitting the same shortcut to close the popup, since `Escape` doesn't work consistently.
- But in Chrome OS (currently Version 70.0.3538.110) hitting that shortcut a second time triggers the notification "Shortcut change. The shortcut to lock the screen has changed. Please use Search+L instead of Ctrl+Shift+L".
- To close PMB at this stage I must either use the mouse (😭) or use the keyboard to navigate to the next tab. Another option would be to select a new keyboard shortcut to activate PMB but I'd rather avoid that if possible.

Any thoughts on implementing `Escape` so that it always closes the PMB popup? Thank you as always!

**Comment by foray1010** [comment]
Now PmB will close by Esc when input has no value, I don't have chrome os, plz help me to test if it works

6.0.1 is released to fix this bug (may have to wait for 1hr to broadcast the release) 🎉 if it doesn't work, plz reopen this issue

**Comment by alon91** [comment]
YES!! 🎉🎉🎉

This is exactly what I was hoping for. Thank you so much for making this happen!!
